data-transformers/transformers/statsTransformer.py

```diff
@@ -144,11 +97,71 @@ def createIODict(a):
         dicts.append(d)
     return dicts
 
+def getFromMinio(url):
+    from commons import getFromUrl
+    return getFromUrl(url)
 
-# Calls Spark
-query = data.map(createEDDict)
-query.saveToCassandra(cassandraKeyspace, "environment_data", ttl=timedelta(hours=1))
-
-query = data.map(createIODict).reduce(lambda a, b: a+b)
-query = sc.parallelize(query)
-query.saveToCassandra(cassandraKeyspace, "io_data", ttl=timedelta(hours=1))
\ No newline at end of file
+def main():
+    from pyspark_cassandra import CassandraSparkContext
+    from pyspark import SparkConf
+    from pyspark import SparkFiles
+    
+    # Takes arguments
+    minioHost = sys.argv[1]
+    filePath = sys.argv[2]
+    trialID = sys.argv[3]
+    experimentID = sys.argv[4]
+    #SUTName = sys.argv[5]
+    containerID = sys.argv[6]
+    hostID = sys.argv[7]
+    statsTable = "environment_data"
+    ioTable = "io_data"
+    alluxioHost = "localhost"
+    
+    # Set configuration for spark context
+    conf = SparkConf().setAppName("Stats Transformer")
+    sc = CassandraSparkContext(conf=conf)
+    
+    # Retrieve the configuration file that was sent to spark
+    #confPath = SparkFiles.get("data-transformers.yml")
+    #with open(confPath) as f:
+    #    transformerConfiguration = yaml.load(f)
+    #    cassandraKeyspace = transformerConfiguration["cassandra_keyspace"]
+    #    minioPort = transformerConfiguration["minio_port"]
+    
+    cassandraKeyspace = "benchflow"
+    minioPort = "9000"
+    
+    #res = urllib2.urlopen("http://"+minioHost+":"+minioPort+"/"+filePath)
+    #compressed = io.BytesIO(res.read())
+    #decompressed = gzip.GzipFile(fileobj=compressed)
+    #lines = decompressed.readlines()
+    lines = getFromMinio("http://"+minioHost+":"+minioPort+"/"+filePath)
+    data = sc.parallelize(lines)
+    
+    activeCpus = 0
+    for l in lines:
+        acpus = 0
+        ob = json.loads(l.decode())
+        for c in ob["cpu_stats"]["cpu_usage"]["percpu_usage"]:
+            if c != 0:
+                acpus += 1
+        if acpus > activeCpus:
+            activeCpus = acpus
+    
+    # Calls Spark
```

Comment:
It's not always clear what exception happens here, so leaving it unspecified for now.